proto/uber/cadence/api/v1/service_domain.proto

```diff
@@ -67,6 +67,7 @@ message RegisterDomainRequest {
   string history_archival_uri = 11;
   ArchivalStatus visibility_archival_status = 12;
   string visibility_archival_uri = 13;
+  string domain_id = 14;
```

Comment:
please add a comment on this field
```
// domain_id is optional. A random UUID is generated if not set. 
// It can be set to a custom UUID as long as it doesn't collide with id of other existing domains
// If unsure leave this empty.
string domain_id = 14;
```